### DISCUSSION TITLE
Time of day restrictions

### DISCUSSION THREAD

**Discussion by nzbnate** [autobrr/autobrr#917] [Ideas]
It would be really nice if it were possible to limit autobrr's ability to grab releases to a specific time window set by the user in the filter settings.

I'm not wanting to use bandwidth during the day (WFH, kids, wife, etc...) so being able to set the window to say 11pm-6am would be great.

**Comment by zze0s** [comment]
Hi! A proper full blown scheduler with time based slots has been on the todo from the start more or less. There's [this item on the project board](https://github.com/orgs/autobrr/projects/1/views/4?pane=issue&itemId=1230601) and there might be something more elsewhere.

There is a plugin for Deluge with a scheduler and that is pretty user friendly to setup. See picture: 
![Pasted image 20220522082132](https://user-images.githubusercontent.com/43699394/236676682-277817d2-ea4c-4fab-9dfc-416598f8f28b.png)

**Reply by stxmqa** [reply]
@zze0s that would mean that the torrent has already been grabbed and is being sent to Deluge. I think @nzbnate intention is to avoid grabbing anything during that time of day.

This is especially important when racing: you don’t want to grab torrents hours after they have been released.

**Reply by ipbeegle** [reply]
AFAIK, most people that are racing are pulling from IRC announce channels, so there is little/no risk (short of a misbehaving bot) of getting old torrents. Maybe during replay from a bouncer? Nevertheless, a new field could be added to the filter 'General' tab to set the max age in seconds since the announce was made.

**Comment by akhdanfadh** [comment]
For anyone coming up to this, you can set a cron job to enable/disable the filter. See the API documentation.

**Comment by ipbeegle** [comment]
For my use case, I wanted limit my racing activity to only grab the popular releases that are announced between 00:00-01:00 AM EST.

I've just been using an external exec filter and sh/test to check if I want a filter to run:

**External filter**
Type: exec
Path to executable: `/bin/sh`
Exec arguments: `-c "export TZ='America/New_York'; [ $(date +%H) -lt 1 ] && [ $(date +%M) -lt 60 ]"`
Expected exit status: `0`

In case people are not familiar with (ba)sh scripts, I've made the following (bash) script:

https://gist.github.com/ipbeegle/20263e2eabeb5c0514050f746b01bf2f

It defaults to using UTC as the time zone, so you'll have to specify your local time zone if you don't want to calculate the offset.

Command-line usage:

```
$ ./autobrr-external-time.sh --help
Usage: ./autobrr-external-time.sh [OPTION]... TIME
Test that the current time is before or after a specified TIME, using an
optionally specified time zone.

TIME should be in HH:MM (24-hour) format. If you do not include the leading '0'
in the hour of your specified TIME, it will be treaded as 12-hour time.
For example:
'1:00' will be treated as 1:00 PM, while '01:00' will be treated as 1:00 AM.

Options:
  -a, --after     Test that the current time is after the specified 'time'.
  -b, --before    (default) Test that the current time is before the
                  specified 'time'.

  --tz, --timezone timezone
    UTC is the default timezone.

    A (probably complete) list of supported time zones is available at:
    https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

Examples:
  Test that the current time in New York, USA (Eastern Time) is before 1:00 AM:

    $ TZ=America/New_York date
    Tue Sep  3 10:05:21 PM EDT 2024
    $ ./autobrr-external-time.sh --tz America/New_York 01:00
    $ echo 0
    1
```

Filter Usage:

(Check that the current time in New York, USA (Eastern Time) is before 1 AM)
**External filter**
Type: exec
Path to executable: `/path/to/autobrr-external-time.sh`
Exec arguments: `--tz America/New_York 01:00`
Expected exit status: `0`

To fulfill the use case that OP posted (Only allow releases between 23:00-06:00), we're unfortunately back to using an embedded script.

**External filter**
Type: exec
Path to executable: `/bin/sh`
Exec arguments: `-c "/path/to/autobrr-external-time.sh --after 23:00 || /path/to/autobrr-external-time.sh 06:00"`
Expected exit status: `0`

Or if you're willing to just do 00:00-06:00:

**External filter**
**External filter**
Type: exec
Path to executable: `/path/to/autobrr-external-time.sh`
Exec arguments: `06:00`
Expected exit status: `0`

Right now it just tests the current time is after/before a specified time and supports time zones. This could be included with the autobrr install until a full scheduler system can be developed.

**Reply by zze0s** [reply]
Nice! We created a new repo recently to collect scripts like these. You're more than welcome to make a PR with it to https://github.com/autobrr/community-scripts

**Reply by ipbeegle** [reply]
Oh that is recent. I'll definitely submit it once I add support for the OP use case.

**Reply by NaruZosa** [reply]
@ipbeegle Sorry to bug you, just wondering if you still plan to change this and submit it?
